Assignment6.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class Assignment6 {
+    /**
+     * Rearranges given to make it the same as desired.
+     *
+     * @param given   initial version of the array of numbers.
+     * @param desired final version of the array of numbers.
+     * @return list of moves made during arrangement.
+     */
+    public static List<List<Integer>> rearrange(int[] given, int[] desired) {
```

Comment:
The return value should be something more precise e.g List<Move> where Move is a simple data structure containing info on a move (car no., slot no. and empty slot no. for instance)